### ISSUE TITLE
A Bookmark Command

### ISSUE THREAD

**Issue by NeilShah2026** [python-discord/sir-lancebot#641]
## Description
When a user tends to use a command often, either by the user manually adding it, or have the bot automatically add through command tracking, add a ".bookmarks", and it will pull up all the bookmarked commands of that user.

## Reasoning
For easier use and memorization of the bot commands.

- [ ] I'd like to implement this feature myself
- [ x] Anyone can implement this feature

**Comment by Brodevil** [comment]
Okay that means a bot  who will book mark all the commands for the user
This will be going to increase the storage and Idea I think it is not good
Because while if any user want to just want to use the command he can just `.help` 
So I don't think this will be a good idea to add .bookmark command

**Comment by Xithrius** [comment]
@NeilShah2026 This bot currently doesn't have a database, such as postres.

I don't think many users would like to be message automatically if the bot noticed a pattern that is set by some constant.

The bot does have a feature for [bookmarking](<https://github.com/python-discord/sir-lancebot/blob/main/bot/exts/evergreen/bookmark.py>) a message, so if the user wants to save something for later, they can always do so.

**Comment by Shivansh-007** [comment]
A possible way for doing this is by going through each message in the user DM history and checking if the message is an embed and if so it has the title of `Bookmark`.

**Comment by Xithrius** [comment]
That's not the best idea if the user has a long message history with the bot.

**Comment by Shivansh-007** [comment]
Agreed, a better way I just remembered as we had discussed in dev-contrib was to pin the bookmark messages in DMs of the user and then the bot could go through the pin history?

**Comment by NeilShah2026** [comment]
True, that would also work

**Comment by Xithrius** [comment]
> I don't think many users would like to be message automatically if the bot noticed a pattern that is set by some constant.

I still think this is relevant. Not sure if the default should have this feature turned on or off.

**Comment by Xithrius** [comment]
I don't want to fill up the user's pinned messages, since it may collide with something in the future.

If anything, we should go with manual additions instead of automatic bookmarking so users aren't randomly DM'd.

**Comment by onerandomusername** [comment]
> @ NeilShah2026 This bot currently doesn't have a database, such as postres.
To note, at this point, the bot *basically* has a database with redis.

**Comment by Xithrius** [comment]
Well, yes, but I'm not sure if it's suited for long-term storage. We'd have to put some time limit on the bookmark or something so we don't use up storage for no reason.
